application/common/models/User.php

```diff
@@ -298,6 +298,15 @@ private function validatePassword(): Closure
             $currentPassword = $this->currentPassword ?? new Password();
             if (! password_verify($this->password, $currentPassword->hash)) {
                 $this->addError($attributeName, 'Incorrect password.');
+            } else {
+                // check the current hash cost and rehash if necessary
+                if (password_needs_rehash(
+                    $currentPassword->hash,
+                    Password::HASH_ALGORITHM,
+                    ['cost' => Password::HASH_COST]
+                )) {
+                    $this->savePassword();
```

Comment:
It looks like this will create a new password record for the User, but won't set it as the current password for that user (unless there is a later call to User->save() that's not obvious from this code, which is possible).